samples/basic/hash_map/sample.yaml

```diff
@@ -36,21 +36,21 @@ tests:
       - CONFIG_SYS_HASH_FUNC32_CHOICE_DJB2=y
   # Newlib
   libraries.hash_map.newlib.separate_chaining.djb2:
-    filter: TOOLCHAIN_HAS_NEWLIB == 1
+    filter: CONFIG_FULL_LIBC_SUPPORTED
     extra_configs:
       - CONFIG_NEWLIB_LIBC=y
```

Comment:
These I really think they should be left 3, TOOLCHAIN_HAS_NEWLIB == 1
Otherwise twister will build them and if newlibc is not supported kconfig will warn and they will be built with whatever they support. That may make the test pass, but it is meaningless, as these tests seem specific for newlib.

Reply:
Thanks. Fixed.